src/app/components/app-settings/app-settings.component.ts

```diff
@@ -37,10 +37,20 @@ export class AppSettingsComponent implements OnInit {
       name: null,
       email: null
     };
+
+    this.auth.loginStatus.subscribe(loggedIn => {
```

Comment:
no they won't, `constructor` and `ngOnInit` are fired every time you open that component. also, please check why `/api/user/details` request is fired multiple times.